nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,161 @@
+"""Site searching Job definition."""
+import collections
+from datetime import datetime, date
+from typing import NamedTuple
+
+from django.conf import settings
+
+from nautobot.extras.jobs import Job
+from nautobot.circuits.models import Circuit
+
+from nautobot_circuit_maintenance.models import CircuitMaintenance
+
+name = "Circuit Maintenance"  # pylint: disable=invalid-name
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_circuit_maintenance", {})
+CIRCUIT_MAINTENANCE_TAG_COLOR = "Purple"
+
+
+class Range(NamedTuple):
```

Comment:
maybe a more specific class name? `DateTimeRange` ?

Reply:
Sure, was keeping with the overflow issue found.